### PR TITLE
Convert sparse labels to pandas series

### PR DESCRIPTION
**Pull request by Neeratyoy**
#### Reference Issue

Addresses #1057.

#### What does this PR implement/fix? Explain your changes.
Adds a check when fetching and converting datasets that converts labels as sparse arrays to dataframes.

#### How should this PR be tested?
```python
import openml
task = openml.tasks.get_task(12731)
_, y = task.get_X_and_y(dataset_format="dataframe")
print(type(y))  # print pandas Series
```

### CODE REVIEW

File: openml/datasets/dataset.py [hunk 1]
```diff
@@ -629,6 +629,8 @@ def _encode_if_category(column):
         elif array_format == "dataframe":
             if scipy.sparse.issparse(data):
                 return pd.DataFrame.sparse.from_spmatrix(data, columns=attribute_names)
+            elif isinstance(data, np.ndarray):
+                return pd.Series(data)
```
Comment (mfeurer) [inline openml/datasets/dataset.py:633]: Hey, what's the reason not to apply `columns=attribute_names` here?

Comment (mfeurer) [reply openml/datasets/dataset.py:633]: Also, how can this become a numpy array in the first place? If I understand this correctly, this is called by `get_data`, which splits off `y` by doing `y = data.iloc[:, targets]` from which point on it should be and stay a series, right?

File: tests/test_tasks/test_task_functions.py [hunk 2]
```diff
@@ -234,3 +234,9 @@ def test_deletion_of_cache_dir(self):
         self.assertTrue(os.path.exists(tid_cache_dir))
         openml.utils._remove_cache_dir_for_id("tasks", tid_cache_dir)
         self.assertFalse(os.path.exists(tid_cache_dir))
+
+    def test_supervised_task_target_format(self):
```
Comment (mfeurer) [inline tests/test_tasks/test_task_functions.py:238]: Why is this a test for a task and not for the respective dataset?

Comment (mfeurer) [reply tests/test_tasks/test_task_functions.py:238]: Also, do we have sufficient tests that the data is loaded correctly for sparse datasets?

Comment (Neeratyoy) [reply tests/test_tasks/test_task_functions.py]: I added this unit test to datasets: https://github.com/openml/openml-python/blob/fix_1057/tests/test_datasets/test_dataset.py#L306

and removed this unit test from tasks as this new unit test makes the one under tasks redundant
I also checked this unit test with the dataset used in the #1057 and this unit test passes for it too as it is a more generalized test given the changes made in this PR

File: tests/test_datasets/test_dataset.py [hunk 3]
```diff
@@ -303,7 +303,22 @@ def test_get_sparse_dataset_with_target(self):
...
+        self.assertEqual(X.shape, (600, 20000))
+
+        self.assertIsInstance(y, pd.Series)
+        self.assertIsInstance(y.dtypes, pd.SparseDtype)
+        self.assertEqual(y.shape, (600,))
+
+        self.assertEqual(len(attribute_names), 20000)
+        self.assertNotIn("class", attribute_names)
+
+    def test_get_sparse_dataset_array(self):
```
Comment (mfeurer) [inline tests/test_datasets/test_dataset.py:321]: As we check for the format of `y` in the new test above, maybe we should also check for the format of `y` here?

File: tests/test_datasets/test_dataset.py [hunk 4]
```diff
@@ -303,7 +303,22 @@ def test_get_sparse_dataset_with_target(self):
...
+    def test_get_sparse_dataset_dataframe_with_target(self):
+        X, y, _, attribute_names = self.sparse_dataset.get_data(
+            dataset_format="dataframe", target="class"
+        )
+        self.assertIsInstance(X, pd.DataFrame)
+        self.assertIsInstance(X.dtypes[0], pd.SparseDtype)
+        self.assertEqual(X.shape, (600, 20000))
+
+        self.assertIsInstance(y, pd.Series)
+        self.assertIsInstance(y.dtypes, pd.SparseDtype)
```
Comment (mfeurer) [inline tests/test_datasets/test_dataset.py:315]: Do you know if scikit-learn can handle `pd.SparseDtype`?

**Review by mfeurer — APPROVED** [review]

**Pull request merged**